modules/battery_manager/jsx/addEntry.js

```diff
@@ -0,0 +1,357 @@
+/**
+ * Battery Manager Add Form
+ *
+ * Module component rendering Add tab
+ *
+ * @author Victoria Foing
+ *
+ */
+class BatteryManagerAddForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      Data: {},
+      formData: {},
+      isLoaded: false,
+      loadedData: 0
+    };
+
+    // Bind component instance to custom methods
+    this.setFormData = this.setFormData.bind(this);
+    this.handleAdd = this.handleAdd.bind(this);
+    this.isDuplicate = this.isDuplicate.bind(this);
+    this.giveOptions = this.giveOptions.bind(this);
+    this.activateEntry = this.activateEntry.bind(this);
+    this.addEntry = this.addEntry.bind(this);
+  }
+
+  componentDidMount() {
+    var self = this;
+    $.ajax(this.props.DataURL, {
+      dataType: 'json',
+      success: function(data) {
+        self.setState({
+          Data: data,
+          isLoaded: true
+        });
+      },
+      error: function(data, errorCode, errorMsg) {
+        console.error(data, errorCode, errorMsg);
+        self.setState({
+          error: 'An error occurred when loading the form!'
+        });
+      }
+    });
+  }
+
+  render() {
+    // Data loading error
+    if (this.state.error !== undefined) {
+      return (
+        <div className="alert alert-danger text-center">
+          <strong>
+            {this.state.error}
+          </strong>
+        </div>
+      );
+    }
+
+    // Waiting for data to load
+    if (!this.state.isLoaded) {
```

Comment:
Can this use @HenriRabalais's new loader component?

Reply:
@driusan doesn't it need to be merged first before we can use it?

Reply:
good point.. I thought it was merged, ignore my comment until it is.